### ISSUE TITLE
run existing old fork of nutch  locally - "hello world"

### ISSUE THREAD

**Issue by rduerr** [amberbiology/nutch-crawler#1]
@rduerr the "BCUBE" technology stack document: https://drive.google.com/file/d/174hEiUJqs4_cwkVaSnAyzDTgva-DZ9ax/view?usp=drive_link refers to the fork of nutch as:

-  https://github.com/b-cube/nutch (forked from the 1.6 branch) 
-  but there is also https://github.com/b-cube/nutch-crawler (which looks like it was copied, rather than forked from the 1.9 branch. 

which is the one that we should be porting, and which was the one that was used in the study? I also noticed that the https://github.com/b-cube/nutch had edits more recently (6 years ago).

**Comment by rduerr** [comment]
@alexlancaster Hmmm.... I think this should have been an issue so just made it one.  Also, I am worried as it didn't email me and unless github isn't configured properly it should have!!!  So it is pure luck that I saw this comment... The answer is https://github.com/b-cube/nutch unless the newer versions of Nutch include all the things we needed for our crawls.  We had to "tailor" nutch pretty substantially to minimize the junk in the crawl frontier and output the good stuff to SOLR.  And are we going to use SOLR again?

**Comment by alexlancaster** [comment]
> @alexlancaster Hmmm.... I think this should have been an issue so just made it one. Also, I am worried as it didn't email me and unless github isn't configured properly it should have!!! So it is pure luck that I saw this comment... 

The project may need configuring to send notifications (I'll check) - or maybe your account might need configuring to send you notifications.

> The answer is https://github.com/b-cube/nutch unless the newer versions of Nutch include all the things we needed for our crawls. We had to "tailor" nutch pretty substantially to minimize the junk in the crawl frontier and output the good stuff to SOLR. 

There is also https://github.com/nsidc/nutch (which most of the documentation refers to). I think it's going to take a while to sort out all the different forks, and try and port the changes to the latest version. I think https://github.com/b-cube/nutch-crawler has more stuff that's newer than https://github.com/b-cube/nutch (since it looks like it's based on 1.9), but unfortunately that version wasn't "forked" - it looks like it was copied into the repo - so it's harder to track it's origin. I may have to do it on the basis of diffs.

We might need to ask the original author to be sure of how it fits together. The docs don't provide enough info as far as I can tell.

> And are we going to use SOLR again?

I don't know - haven't got that far yet.

**Comment by alexlancaster** [comment]
under settings, this is how I have mine setup - this should send emails for "@-style" mentions:



![Image](https://github.com/amberbiology/nutch-crawler/assets/5281946/c4c36aed-68e4-4403-8ebc-7d1831b08f62)

**Comment by rduerr** [comment]
I'll check my settings; but all the other repos I'm in have done that automatically, so that is wierd.

I note that I have already emailed Beto (Luis) about this.  I hope to hear back soon.  He is still at NSIDC, so can't be contracted to through his normal employment.  I had actually tried to get him earlier but it turned out he was on vacation earlier this month, so... oh well...

**Comment by rduerr** [comment]
Hah!  That's exactly how I have mine set up!!! Soo... no explanation yet.

**Comment by alexlancaster** [comment]
No worries. I don't think it'll hold us up making progress, I can figure out a lot just from context and comparing repository branches. It might be nice to get a confirmation before we really get stuck into writing any new code. Right now, my main goal is to port the changes made against the original 1.9 branch to the latest upstream version, and make sure they run.

I did also attempt to build the current repo as-is using `ant`, but, as I suspected, the dependencies are too old (many of them don't exist any more). So it can't be built as-is, so it'll definitely need a port to the latest upstream apache.

**Comment by rduerr** [comment]
and now is about when that paper Luis wrote becomes extremely important...

**Comment by alexlancaster** [comment]
e.g. running `ant clean runtime`, generates errors like:

```
[ivy:resolve] 	  http://repo1.maven.org/maven2/com/google/code/crawler-commons/crawler-commons/0.4/crawler-commons-0.4.jar
[ivy:resolve] 		::::::::::::::::::::::::::::::::::::::::::::::
[ivy:resolve] 		::          UNRESOLVED DEPENDENCIES         ::
[ivy:resolve] 		::::::::::::::::::::::::::::::::::::::::::::::
[ivy:resolve] 		:: org.slf4j#slf4j-api;1.6.1: not found
[ivy:resolve] 		:: org.slf4j#slf4j-log4j12;1.6.1: not found
[ivy:resolve] 		:: org.mockito#mockito-all;1.8.4: not found
[ivy:resolve] 		:: org.hamcrest#hamcrest-all;1.1: not found
[ivy:resolve] 		:: log4j#log4j;1.2.15: not found
[ivy:resolve] 		:: commons-lang#commons-lang;2.6: not found
[ivy:resolve] 		:: commons-collections#commons-collections;3.1: not found
[ivy:resolve] 		:: commons-httpclient#commons-httpclient;3.1: not found
[ivy:resolve] 		:: commons-codec#commons-codec;1.3: not found
[ivy:resolve] 		:: org.apache.hadoop#hadoop-core;1.2.0: not found
[ivy:resolve] 		:: org.apache.tika#tika-core;1.5: not found
[ivy:resolve] 		:: com.ibm.icu#icu4j;4.0.1: not found
[ivy:resolve] 		:: xerces#xercesImpl;2.9.1: not found
[ivy:resolve] 		:: xerces#xmlParserAPIs;2.6.2: not found
[ivy:resolve] 		:: oro#oro;2.0.8: not found
[ivy:resolve] 		:: com.google.guava#guava;11.0.2: not found
[ivy:resolve] 		:: com.google.code.crawler-commons#crawler-commons;0.4: not found
[ivy:resolve] 		::::::::::::::::::::::::::::::::::::::::::::::
[ivy:resolve] :::: ERRORS
```

I am able to build the upstream nutch (w/o the mods, obviously), however.

**Comment by alexlancaster** [comment]
> Hah! That's exactly how I have mine set up!!! Soo... no explanation yet.

odd. next time it makes sense, try and tag me with @alexlancaster to see if it generates a notification to me.  it might work if I assign the card to you, maybe

**Comment by rduerr** [comment]
Found it - it turns out that github notifications were masquerading as a person called Jen Preciado.  WHy?  I don't know... but I was ignoring them as I don't know a Jen Preciado...  But I forced a fix in my contacts...

**Comment by rduerr** [comment]
Using the newer version is good; but it may well mean that we need to either re-develop or re-deploy some of the patches Luis had created to take care of what got onto the crawl frontier and what ended up in SOLR.  But I think that is OK, as we likely need to update both bits anyway.

**Comment by alexlancaster** [comment]
Agreed. Yes, that's my goal, is to isolate the changes that were made relative to vanilla 1.9, and then create a patch that I will then re-apply to the most recent upstream.  I suspect the patch won't apply cleanly, but it should be possible to modify it so that will apply. 

It would be nice if it were possible to run the old crawler as-is to have a baseline comparison, but I think the version drift with the dependencies is to just too much for that to work.  Plus, at the end of the day, we don't want to be developing against an ancient version of nutch.

**Comment by rduerr** [comment]
Here is Luis' reply to my request:

I think at the end [we could use 1.9](https://github.com/b-cube/nutch-crawler) so I'd start there. The reason to copy the repo instead of forking it was just to work with a cleaner git history as the original had several big files on them. I vaguely remember that Siri Jodha wanted to do some crawls for EGU and I think that was related to RDA and that branch.

I bet a lot has changed since we last worked on this project, one thing it would be useful is to find a way to persist the state of the crawl to S3 (the crawldb store in HDFS) so we can start from where we left it. Another thing, we never used fancy ML filters, we used a filter with a few rules for controlling the crawling frontier and another to rank how relevant a service was(once we found it) based on regexes and some custom logic. Hope it's not too complicated to dive into, you know software is tricky after a few years. Let me know how it goes,
